lib: multicell_location: Remove device id (#5881)
nrfconnect/sdk-nrf: doc/nrf/releases/release-notes-changelog.rst

  * :ref:`multicell_location` sample:

    * Updated to only request neighbor cell measurements when connected and to only copy neighbor cell measurements if they exist.

>     * Remove device ID from :c:func:`multicell_location_get` parameter list. nRF Cloud and HERE didn't use it. Skyhook will now set modem UUID as it's device ID.

Reviewer: ```suggestion
    * Removed device ID from the :c:func:`multicell_location_get` parameter list. nRF Cloud and HERE did not use it. Skyhook will now set modem UUID as its device ID.
```
